tls/s2n_kem.c

```diff
@@ -467,14 +480,15 @@ bool s2n_kem_is_available(const struct s2n_kem *kem)
         return false;
     }
 
-    bool available = s2n_libcrypto_supports_evp_kem();
+    if (kem->kem_nid == S2N_NID_MLKEM768 || kem->kem_nid == S2N_NID_MLKEM1024) {
+        return s2n_libcrypto_supports_mlkem();
+    }
 
-    /* Only newer versions of libcrypto have ML-KEM support. */
-    if (kem == &s2n_mlkem_768) {
-        available &= s2n_libcrypto_supports_mlkem();
+    if (kem->kem_nid == S2N_NID_KYBER512 || kem->kem_nid == S2N_NID_KYBER768 || kem->kem_nid == S2N_NID_KYBER1024) {
+        return s2n_libcrypto_supports_evp_kem();
     }
 
-    return available;
+    return false;
```

Comment:
Given that we already initialize `kem->nid` [based on the feature probe](https://github.com/aws/s2n-tls/blob/main/tls/s2n_kem.h#L34-L48), do we even need to check the feature probe at runtime here? Like couldn't we just use the `kem->nid == NID_undef` check above, and not make the user update `s2n_kem_is_available` at all when adding a new kem?

So just this?
```
if (kem == NULL || kem->kem_nid == NID_undef) {
    return false;
}
return true;
```